src/ast/passes/return_path_analyser.h

```diff
@@ -1,32 +1,27 @@
 #pragma once
 
 #include "ast/pass_manager.h"
-#include "ast/visitors.h"
+#include "ast/visitor.h"
 
 namespace bpftrace {
 namespace ast {
 
-class ReturnPathAnalyser : public Dispatcher<bool> {
+class ReturnPathAnalyser : public Visitor<ReturnPathAnalyser, bool> {
 public:
   explicit ReturnPathAnalyser(ASTContext &ctx, std::ostream &out = std::cerr);
 
   // visit methods return true iff all return paths of the analyzed code
   // (represented by the given node) return a value
   // For details for concrete node type see the implementations
-  bool visit(Program &prog) override;
-  bool visit(Subprog &subprog) override;
-  bool visit(Jump &jump) override;
-  bool visit(If &if_stmt) override;
-
-  // For nodes that neither directly return a value or have children with
-  // semantics such that the code represented by the node always returns
-  // a value, false is returned
-  bool default_visitor(Node &node) override;
+  using Visitor<ReturnPathAnalyser, bool>::visit;
```

Comment:
Just for my understanding, this brings into scope all the overloaded `visit()` methods, right?

Reply:
Right. In C++ if you declare one method with the same name, then it immediately shadows all other methods unless there is an explicit using statement.